### ISSUE TITLE
✅ 🔗 🐞 Container gap values doesn't persist on copy pasting between two sites (Duplicate of #24111)

### ISSUE THREAD

**Issue by AF1QUE** [elementor/elementor#28343]
### Prerequisites

- [ ] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

Container gap values (when unlinked) doesn't get copied over to another site properly. If the values of column and row are different (which makes me unlink the fields) then the value will not persist when I copy the section to another website.

![origin](https://github.com/user-attachments/assets/580da644-ef3d-446c-898f-1f1c704c99cc)

![destination](https://github.com/user-attachments/assets/d4708f12-ff61-4196-bca1-ad02e907e163)

### Steps to reproduce

1. Create a section using container having different column and row gap values, or keep gap fields unlinked.
2. Copy the section 
3. Paste the section on another elementor website from "Paste from another site"

### Expected behavior

After pasting the copied section the gap values should persist or not reset to 0, currently it resets to 0 if the gap fields are unlinked. 

### Isolating the problem

- [ ] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [X] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt


== Server Environment ==
	Operating System: Linux
	Software: Apache/2.4.57 (Ubuntu) mod_fcgid/2.3.9 OpenSSL/1.1.1f
	MySQL version: mariadb.org binary distribution v10.11.4-MariaDB-1:10.11.4+maria~ubu2004
	PHP Version: 8.0.29
	PHP Memory Limit: 128M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 256M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.6.1
	Site URL: https://darling-nightingale-e5592b.instawp.xyz
	Home URL: https://darling-nightingale-e5592b.instawp.xyz
	WP Multisite: No
	Max Upload Size: 256 MB
	Memory limit: 128M
	Max Memory limit: 128M
	Permalink Structure: /%postname%/
	Language: en_US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 3.1.0
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en-US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 3.23.3
		Author: Elementor.com

	Elementor Pro
		Version: 3.23.1
		Author: Elementor.com

	JetElements For Elementor
		Version: 2.6.19
		Author: Crocoblock


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Improved CSS Loading: Active by default
	Inline Font Icons: Active by default
	Additional Custom Breakpoints: Active by default
	Flexbox Container: Active by default
	Grid Container: Active by default
	Upgrade Swiper Library: Active by default
	Nested Elements Performance: Active by default
	Optimized Control Loading: Active by default
	Optimized Markup: Inactive by default
	Plugin Onboarding: Active by default
	Default to New Theme Builder: Active by default
	Header & Footer: Active by default
	Elementor Home Screen: Active by default
	Editor Top Bar: Active by default
	Build with AI: Active by default
	Landing Pages: Inactive by default
	Nested Elements: Active
	Lazy Load Background Images: Active by default
	Pages Panel: Inactive by default
	Element Caching: Active by default
	Link In Bio: Active by default
	Floating Buttons: Active by default
	Elementor Editor Events: Inactive by default
	Display Conditions: Active by default
	Form Submissions: Active by default
	Menu: Active
	Taxonomy Filter: Active by default
	Search: Active by default


== Log ==
	
Log: showing 5 of 52024-07-28 13:31:04 [info] elementor-pro::elementor_pro_updater Started 
2024-07-28 13:31:04 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-07-28 13:31:04 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-07-28 13:31:04 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.23.0',
  'to' => '3.23.1',
)]
2024-07-28 13:31:04 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.23.0',
  'to' => '3.23.1',
)]

PHP: showing 2 of 2PHP: 2024-07-29 05:42:30 [warning X 6][../wp-content/plugins/elementor/includes/elements/container.php::1835] Undefined array key "size" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-08-06 21:23:14 [warning X 13][../wp-content/plugins/elementor/includes/managers/image.php::113] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 2 of 2JS: 2024-07-29 13:16:44 [error X 1][../wp-admin/load-scripts.php?c=0&loadchunk_0=jquery-core,jquery-migrate,utils&ver=6.6.1:2:28760] Cannot read properties of undefined (reading 'value') 
JS: 2024-08-07 22:14:29 [error X 1][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.23.3:3:663451] Cannot read properties of undefined (reading 'isDesignable') 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Incompatible
	JetElements For Elementor: Incompatible

== Elementor Pro - Compatibility Tag ==
	
	JetElements For Elementor: Incompatible

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by Daveden2** [comment]
There's already a [bug report](https://github.com/elementor/elementor/issues/24111) for it I believe.

**Comment by AF1QUE** [comment]
> There's already a [bug report](https://github.com/elementor/elementor/issues/24111) for it I believe.

Thank you letting me know. I see, this has been issue a for a long time now. I think this needs to be addressed as soon as possible.

**Comment by afnanmishu** [comment]
I've been facing this problem for a long time! 

It's quite frustrating to manually adjust the gap each time I paste a single widget or component onto a new site. Elementor should fix this bug as soon as possible. @elementorbot

**Comment by KingGrizzly** [comment]
I haven't found a fix other than cloning a starter site with my webhost for new builds.

A demonstration of the bug precisely between two fresh environments here: https://www.loom.com/share/adda9c50442b4135b2ab10dec759d42b?sid=852315b8-0483-4716-b76a-b99487d6110e

I've pointed out the problem to Elementor Support who reproduced and acknowledge the bug then went silent. When I raised the issue out of desperation on Elementor Facebook, I was scolded for doing so. This is a major flaw and a promoted feature which has been broken ever since Elementor had trouble moving from a single gap value to separate row and column gap values. 

Elementor's shorthand CSS rendering of the two gap values is another issue in its own right, and likely related to the cross site pasting and export/import issues: https://github.com/elementor/elementor/issues/24178 ). Here's why this is a problem:

Row and column gap values get written in Elementor CSS as:

elementor-element {
--gap: 18px 12px;
}

They are stuck together instead of broken out as --gap-column and --gap-row values ([these are standard](https://css-tricks.com/almanac/properties/g/gap/)), which renders them unusable.

Let's say a developer wanted to do some CSS calc against the row or column gap value. Since --gap-column
--gap-row don't exist, one is left with the --gap value which is actually both values (eg: 18px 12px). Thus, something like the following fails:

calc(var(--gap) * 2) because it is calc(18px 12px * 2) which makes no sense and doesn't allow a developer to target the row or column gaps individually.

This cripples the utility of the pencil icon for custom units and prevents developers from creating efficiencies with their own variables and helper classes.

If Elementor was able to keep up with Github, this feedback would have been digested and a solution implemented months ago. One person appears to be tasked with working on Github, and facilitating Facebook, and facilitating Discord! How is this feasible? It's a major red flag, and this critique is coming from a massive Elementor fan. I want Elementor to thrive and offer these posts as a supporter.

My agency had a solution for the Elementor community a year ago which we planned to release for free and have been unable to because no one would be able to import our containers properly.

**Comment by captainplod** [comment]
Thanks for the explanation and detailed investigation @KingGrizzly - learnt a few things there! Very disappointing that this has not been resolved by Elementor yet as my only solution right now is to use a gap system of using the same px for columns and rows i.e 30px and 30px for sections that are being copied over to new websites. Huge Elementor fan too from the early days - I just wish they'd fix bugs like this rather than releasing new features

**Comment by nicholaszein** [comment]
Hello everyone!

Thank you for your submission! 🙏 

### 🔗 🐞 This issue is a Duplicate of #24111

Which is on the same subject. 

For this reason, I'm closing this thread and 🔃 **Merging** it with the original, so we can have all the discussions in one location, and deal with it more efficiently.

Feel free to continue the conversation there. 😉 

### 🔍 Please, in the future, remember to check if there isn't any thread similar to yours already opened. 

Kind regards

**Comment by nicholaszein** [comment]
Hey everyone!

We have great news! 🙌 

### 📢 We're happy to announce that this issue was resolved with the release of Elementor v3.26.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
